lib/puppet/face/config.rb

```diff
@@ -163,7 +163,13 @@ def report_section_and_environment(section_name, environment_name)
       Puppet::FileSystem.touch(path)
       Puppet::FileSystem.open(path, nil, 'r+:UTF-8') do |file|
         Puppet::Settings::IniFile.update(file) do |config|
-          config.set(options[:section], name, value)
+          if options[:section] == "master"
+            config.set(options[:section], name, value)
+            # set the same value to the server section when master settings are set
+            config.set("server", name, value)
+          else
+            config.set(options[:section], name, value)
+          end
```

Comment:
not sure if this is the right approach as it can complicates things managing bot [server] and [master] sections in the puppet.conf file.

Reply:
Yeah I'm thinking if the user specified `--section master` then we should only write it to the `[server]` section, and then when a setting is read, read from `server`, `master`, `main` sections. That part isn't working currently though, see my comment in https://tickets.puppetlabs.com/browse/PUP-10671?focusedCommentId=762917&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-762917. Assuming that is fixed, then I think {{puppet config}} only needs to handle the case where someone tries to set or delete a setting in the {{master}} section.